Header header
string[] object_names